### PR TITLE
Keybindings: Adds `open` keyboard shortcut

### LINKED ISSUE #4142: Keybindings: Add keyboard shortcut to open a file
**Is your feature request related to a problem? Please describe.**
Add a keyboard shortcut for opening a file. The keyboard shortcut is `ctrl+o`.

**Describe the solution you'd like**
First, follow the steps used in PR #4138 or #4139 for hooking up the keyboard bindings. Next, an `action` and `reducer` will need to be created for opening a file. Finally, we'll need to develop an interface for opening a new file or redirect the user to the `directory` view.

**Describe alternatives you've considered**
N/A

**Additional context**
Pulled from issue #4104.

### PR DESCRIPTION
**Pull request by stormpython**
Adds the `open` keyboard shortcut to nteract. Also wraps `mapDispatchToProps` with the function 
`makeMapDispatchToProps`.

This closes #4142.

### CODE REVIEW

**Comment by stormpython** [comment]
Currently this branch is broken. While the keybinding works, it causes an error in the console.

screen shot 2019-01-30 at 2 12 26 pm

File: applications/jupyter-extension/nteract_on_jupyter/app/contents/index.tsx [hunk 1]
```diff
@@ -128,6 +129,7 @@ const makeMapStateToProps: any = (
 
   const appBase: string = urljoin(host.basePath, "/nteract/edit");
 
+  // tslint:disable-next-line typedef
```
Comment (captainsafia) [inline applications/jupyter-extension/nteract_on_jupyter/app/contents/index.tsx]: I think you can remove this disable and add a return type to `mapStateToProps` that defines the schema for object returned.

File: applications/jupyter-extension/nteract_on_jupyter/app/contents/index.tsx [hunk 2]
```diff
@@ -164,79 +166,82 @@ const makeMapStateToProps: any = (
...
-        actions.restartKernel({
-          outputHandling,
-          contentRef: initialProps.contentRef
-        })
-      );
-    },
-    SAVE: () => dispatch(actions.save({ contentRef: initialProps.contentRef }))
-  }
-});
+// tslint:disable-next-line typedef
```
Comment (captainsafia) [inline applications/jupyter-extension/nteract_on_jupyter/app/contents/index.tsx]: Ditto here. I think you can make the return type `(dispath: Dispatch, ownProps: { appBase: string; contentRef: ContentRef }): object` for this function.

**Review by captainsafia — APPROVED** [review]
Looks good overall! I left a few type-related comments inline.

File: packages/actions/src/actions/contents.ts [hunk 3]
```diff
@@ -101,6 +101,14 @@ export function downloadContentFulfilled(payload: {
   };
 }
 
+export function open(url: string): actionTypes.Open {
+  // Redirect to directory view in new tab
+  window.open(url, "_blank");
```
Comment (rgbkrk) [inline packages/actions/src/actions/contents.ts]: I'm a little nervous about this being a side effect in an action creator. I think we can just put the window.open call into the menu directly as a special case. If we still want to log something / have it noticed in the reducers, we can dispatch this still, just without the side effect.

Comment (stormpython) [reply packages/actions/src/actions/contents.ts]: Good point.

File: applications/jupyter-extension/nteract_on_jupyter/app/contents/index.tsx [hunk 4]
```diff
@@ -215,6 +217,7 @@ const mapDispatchToProps = (
           restarting: false
         })
       ),
+    OPEN: () => window.open("/nteract/edit", "_blank"),
```
Comment (rgbkrk) [inline applications/jupyter-extension/nteract_on_jupyter/app/contents/index.tsx]: Just realized this should be ``window.open(urljoin(appBase, "nteract/edit"), "_blank")`` or something similar to that.

**Comment by rgbkrk** [comment]
Is this still WIP or is it ready for a review?

**Comment by stormpython** [comment]
@rgbkrk, I think I changed it to [WIP] because of failing tests. Let me take another look tonight, and if its ready, I will remove the `WIP` label.

**Comment by stormpython** [comment]
@rgbkrk, this ready for review.

**Review by rgbkrk — APPROVED** [review]
Awesome.

**Pull request merged**